streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/processors/ThreadLocalProcessor.kt

```diff
@@ -0,0 +1,101 @@
+package com.mirego.trikot.streams.reactive.processors
+
+import com.mirego.trikot.foundation.concurrent.MrFreeze
+import com.mirego.trikot.foundation.concurrent.dispatchQueue.DispatchQueue
+import com.mirego.trikot.streams.cancellable.CancellableManager
+import com.mirego.trikot.streams.reactive.observeOn
+import com.mirego.trikot.streams.reactive.subscribe
+import com.mirego.trikot.streams.reactive.subscribeOn
+import org.reactivestreams.Processor
+import org.reactivestreams.Publisher
+import org.reactivestreams.Subscriber
+import org.reactivestreams.Subscription
+import kotlin.native.concurrent.ThreadLocal
+
+class ThreadLocalProcessor<T>(
+    private val parentPublisher: Publisher<T>,
+    private val observeOnQueue: DispatchQueue,
+    private val subscribeOnQueue: DispatchQueue
+) : Processor<T, T> {
+    override fun onSubscribe(s: Subscription) {
```

Comment:
Can't quite wrap my head around it: Why not extends AbstractPublisher

Reply:
What AbstractProcessor gives us in this case ?